src/cli/self_update/windows.rs

```diff
@@ -916,71 +867,64 @@ mod tests {
         ];
         let cargo_home = wide(r"c:\users\example\.cargo\bin");
         let final_path = [&cargo_home, &[b';' as u16][..], &initial_path].join(&[][..]);
+        let final_path = HSTRING::from_wide(&final_path);
```

Comment:
@kennykerr do you think it would make sense to add `push_str()` to `HSTRING`?

Reply:
That would need to be on `HStringBuilder`. A `HSTRING` is an immutable string (more or less the equivalent of `Box<[u16]>` but without mutability) whereas `HStringBuilder` is a `Vec` equivalent.

Reply:
Ah, I hadn't noticed `HStringBuilder` yet -- so the type is called `HSTRING` but the builder is called `HStringBuilder`?

So I guess maybe we should be using a `HStringBuilder` here instead of joining `u16` slices?